graphql_compiler/global_utils.py

```diff
@@ -32,6 +49,14 @@ class ASTWithParameters:
     query_ast: DocumentNode
     parameters: Dict[str, Any]
 
+    @classmethod
+    def from_query_string_with_parameters(
```

Comment:
How about we make a `to_query_string_with_parameters` instance method in this class and a `to_ast_with_parameters` method in the `QueryStringWithParameters`? That way the conversion would be more concise and we are less likely to go over the 100 character limit when typing it out.

Reply:
With a `to_` instance method, both the "source" and the "destination" types are pinned down at definition time (source type by the `self` type, destination by the return type). This means the caller at the call site has no freedom to pick the destination type other than by looking for an appropriate `to_` method. If such a method doesn't exist, the caller is just out of luck.

With a `from_` classmethod, the call site controls the target type, because the call site does `Foo.from_x()` and `Foo` becomes the `cls` argument. This not only gives us more flexibility and convenience, but also means that we get more functionality out of fewer from-to function pairs.